### ISSUE TITLE
Attribution & Contribution reporting

### ISSUE THREAD

**Issue by tprieur** [WICG/attribution-reporting-api#1356]
It is common practice in the adtech industry to offer attribution and contribution reporting.

- Attribution typically refers to the Campaign that received the final touchpoint, or any other attribution model utilized by the adtech.
- Contributing campaigns are those campaigns with which the user engaged, but did not receive the attribution.

An attribution model like last touch tends to benefit Re-marketing and Search campaigns, while contribution emphasizes the positive influence of Awareness campaigns such as Video or Branding campaigns.

As outlined in [this resource](https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#multiple-sources-for-the-same-trigger-multi-touch), it appears that the Attribution Reporting API (ARA) currently lacks the capability to provide contribution reporting. This limitation could significantly impact branding campaigns and potentially introduce a bias towards lower funnel campaigns.

Could you kindly confirm whether ARA supports Contribution reporting?

Additionally, could you provide guidance on potential workarounds for overcoming ARA limitations in scenarios like these? Shared Storage could be used for such use cases, but is highly impractical due to the budgeting drawbacks (such as shared budget for all sources & advertiser).

**Comment by jkarlin** [comment]
Curious to hear more on the practicality (or lack thereof) of using Shared Storage for scenarios like this. The concern you raise is shared budget, but I'd like to understand how impractical it actually is. For instance, you could create a queue of reports in shared storage and send them out over time as budget allows. This could help if you're worried about bursty traffic from a single user. If conversion data is constant then I suppose not.

**Comment by alois-bissuel** [comment]
Hello @jkarlin,

To the best of my knowledge, there is no way of knowing whether there is budget left in Private Aggregation API. Hence queuing reports would just be delaying the issue and hoping that there is budget remaining at that point.

**Comment by jkarlin** [comment]
That's correct, but given that the budget resets frequently, it seems probable that you'll be able to run them in the future, no?
